packages/@tailwindcss-upgrade/src/index.ts

```diff
@@ -230,18 +230,31 @@ async function run() {
     }
 
     info('Updating dependencies…')
-    for (let dependency of [
-      'tailwindcss',
-      '@tailwindcss/cli',
-      '@tailwindcss/postcss',
-      '@tailwindcss/vite',
-      '@tailwindcss/node',
-      '@tailwindcss/oxide',
-      'prettier-plugin-tailwindcss',
-    ]) {
+    {
+      let dependencies = (
+        await Promise.all(
+          [
+            'tailwindcss',
+            '@tailwindcss/cli',
+            '@tailwindcss/postcss',
+            '@tailwindcss/vite',
+            '@tailwindcss/node',
+            '@tailwindcss/oxide',
+            'prettier-plugin-tailwindcss',
+          ].map(async (dependency) => {
+            if (dependency === 'tailwindcss') {
+              return dependency
+            } else if (await pkg(base).has(dependency)) {
```

Comment:
could you hoist `await pkg(base)` outside of this call? Then you don't need the promise.all and things get simpler again?

Reply:
Only if we change how `pkg` works. Right now `pkg(base)` itself just returns an object with `async add()`, `async has()` and `async remove()` methods.

Reply:
We could also make it sync